src/wit_gen.rs

```diff
@@ -9,7 +9,7 @@ use crate::{
 
 /// A struct that represents a witness generation context. It provides an interface for assigning
 /// values to witness columns in a circuit.
-#[derive(Debug, Default)]
+#[derive(Debug, Default, Clone)]
```

Comment:
@lancenonce I think this is the last thing, we should remove Default here, because we don't want people creating this with num_steps=0 by mistake.

Reply:
So we are still waiting for this

Reply:
Just committed the change, msgd you a few questions about this but this should be correct.

Reply:
Thanks, I just realised that my comment was in the wrong line, but you did the right thing.